tpu-client-next/src/connection_workers_scheduler.rs

```diff
@@ -216,14 +217,76 @@ impl ConnectionWorkersScheduler {
             max_reconnect_attempts,
             leaders_fanout,
         }: ConnectionWorkersSchedulerConfig,
+        broadcaster: Broadcaster,
     ) -> Result<Arc<SendTransactionStats>, ConnectionWorkersSchedulerError> {
-        let ConnectionWorkersScheduler {
+        ConnectionWorkersSchedulerBuilder::new(
+            self.leader_updater,
+            self.transaction_receiver,
+            self.update_identity_receiver,
+            ConnectionWorkersSchedulerConfig {
+                bind,
+                stake_identity,
+                num_connections,
+                skip_check_transaction_age,
+                worker_channel_size,
+                max_reconnect_attempts,
+                leaders_fanout,
+            },
+            self.cancel,
+        )
+        .run(broadcaster)
+        .await
+    }
+}
+
+pub struct ConnectionWorkersSchedulerBuilder {
+    leader_updater: Box<dyn LeaderUpdater>,
+    transaction_receiver: mpsc::Receiver<TransactionBatch>,
+    config: ConnectionWorkersSchedulerConfig,
+    update_identity_receiver: watch::Receiver<Option<StakeIdentity>>,
+    cancel: CancellationToken,
+}
+
+impl ConnectionWorkersSchedulerBuilder {
+    pub fn new(
+        leader_updater: Box<dyn LeaderUpdater>,
+        transaction_receiver: mpsc::Receiver<TransactionBatch>,
+        update_identity_receiver: watch::Receiver<Option<StakeIdentity>>,
```

Comment:
If i make `update_identity_receiver` to be option, the logic in the select will become really cumbersome.